wsmaster/codenvy-hosted-api-organization/src/main/java/com/codenvy/organization/api/resource/OrganizationalAccountAvailableResourcesProvider.java

```diff
@@ -0,0 +1,184 @@
+     */
+    private List<Resource> min(Collection<? extends Resource> resources1, Collection<? extends Resource> resources2) {
+        final Map<String, Resource> typeToResource1 = resources1.stream()
+                                                                .collect(toMap(Resource::getType,
+                                                                               Function.identity()));
+        List<Resource> result = new ArrayList<>();
+        for (Resource resource2 : resources2) {
+            String resourceType = resource2.getType();
+            Resource resource1 = typeToResource1.get(resourceType);
+            if (resource1 != null) {
+                if (resource1.getAmount() < resource2.getAmount()) {
+                    typeToResource1.put(resourceType, resource1);
+                } else {
+                    typeToResource1.put(resourceType, resource2);
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Returns resources which are used by suborganizations of specified organization.
+     *
+     * <p>Note that the result will includes used resources of all direct and nested suborganizations.
+     *
+     * @param parentQualifiedName
+     *         parent qualified name, e.g. 'parentName/suborgName
+     * @return resources which are used by suborganizations of specified organization.
+     * @throws ServerException
+     *         when any other exception occurs on calculation of used resources
+     */
+    private List<Resource> getUsedResourcesBySuborganizations(String parentQualifiedName) throws ServerException {
+        ResourceUsageManager resourceUsageManager = resourceUsageManagerProvider.get();
+        List<Resource> usedResources = new ArrayList<>();
+        for (Organization suborganization : Pages
+                .iterate((maxItems, skipCount) -> organizationManager.getSuborganizations(parentQualifiedName,
+                                                                                          maxItems,
+                                                                                          skipCount))) {
+            try {
+                // make unavailable for parent shared resources that are used
+                usedResources.addAll(resourceUsageManager.getUsedResources(suborganization.getId()));
+            } catch (NotFoundException e) {
+                throw new ServerException(e.getLocalizedMessage(), e);
+            }
+        }
+        return new ArrayList<>(resourceAggregator.aggregateByType(usedResources)
+                                                 .values());
+    }
+
+    /**
+     * Returns formatted string for list of resources.
+     */
+    private static String format(Collection<? extends Resource> resources) {
+        return '[' +
+               resources.stream()
+                        .map(resource -> resource.getAmount() + resource.getUnit() + " of " + resource.getType())
```

Comment:
There is necessity to have resources formatter that will return readable string for resource with converting to unit that is the most suitable, e.g. instead of 1024 mb of RAM -> 1Gb of RAM.
So it will be implemented later, and then we'll remove this private method